File: array_rotation.php
<?php

function rotateLeft($d, $arr)
{
    // Write your code here
    $len = count($arr);
    $rotations = ($d > $len ? ($len % $d) + 1 : $d);
    $rotated = array_slice($arr, 0, $rotations);
    $suffix_array = array_slice($arr, $rotations);
    return array_merge($suffix_array, $rotated);
}

print_r([1,2,3,4,5], 3);

File: array_rotation.py
#given array a of size n, rotate it d time to the left

def rotLeft(a, d):
    # Write your code here
    for counter in range(d):
        a.append(a.pop(0))
    return a


print(rotLeft([33, 47, 70, 37, 8, 53, 13,
      93, 71, 72, 51, 100, 60, 87, 97], 13))
print(rotLeft([1, 2, 3, 4, 5], 4))
